components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java

```diff
@@ -112,7 +112,7 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
 
         if (StringUtils.isEmpty(userStoreDomain)) {
             context.setSubject(
-                    AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(authenticatedUserName));
+                    AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(fullyQualifiedName));
```

Comment:
Case: I have 2 domains (domain1 with user bob@domain1, domain2 with user bob@domain2). In IS created 2 IDP (idp-domain-1, idp-domain-2). I configured my SP for authentication on IS by SAML (using idp-domain-1 or idp-domain-2).  After recieving SAML response with username ('bob') only, i can't  uniquely identify which 'bob' was authenticated.